### ISSUE TITLE
Automatically building nuspec files for csproj files

### ISSUE THREAD

**Issue by AwaNoodle** [lholman/OneBuild#27]
I think it would be pretty good to make use of the nuget feature to generate nuspec files and pack them based off a csproj. Atm I have to roll-my-own nuspec file for every output that I want. This stops me using a common workflow where I can call **nuget spec** in the same folder as the csproj to create my nuspec (with all the tags to use description and id from the project) and then calling **nuget pack <csproj name>** to create my nuget file. It saves me having to track every output I need. 

Since it follows a convention that the nuspec will sit next to the csproj, could we have a feature to check for a nuspec sitting next to the csproj and then have OneBuild call pack on the csproj rather than the nuspec file?

**Comment by lholman** [comment]
Yes, that seems fairly trivial to add to the New-Packages task and associated module. I can see the value if you're heavily leaning on NuGet packaging.

I see it adding to the current convention http://lholman.github.io/OneBuild/conventions.html that does the following:

"Generates new Nuget ([packageName].[version].nupkg) and optional Symbols({packageName].[version].symbols.nupkg) package(s) by passing all .nuspec files found in the solution root folder."

With the addition of..

"Generates new Nuget ([packageName].[version].nupkg) and optional Symbols({packageName].[version].symbols.nupkg) package(s) by identifying any .nuspec files under the solution root that contain the name of and are at the same level as a Visual Studio project file (.csproj..and?) with a corresponding name and then executing Nuget Pack for the VS project files."

A few thoughts..
- What project types should be supported?
- Should we support symbol packages too, as OneBuild does now?

**Comment by AwaNoodle** [comment]
I think csproj is the most likely to be used so I would ensure that is working. However, if you are scanning and matching on csproj adding a vbproj (if supported) isn't likely to cause a problem. 

I like the idea of the symbols too. If someone has added the a symbols nuspec sitting next to the csproj then that seems a nice way to opt in.

**Comment by samshiles** [comment]
Another option would be to support OctoPack for projects which have the target installed by specifying /p:RunOctoPack=true when invoking the compilation of the solution.

**Comment by lholman** [comment]
@AwaNoodle Does the merging of #36 (originally #31) solve this for you, if so, could you close?

**Comment by AwaNoodle** [comment]
Not fully. I won't always be using Octopack.

**Comment by lholman** [comment]
Ok, let's get early visibility of the PR so it's not a big surprise. 

ll

-----Original Message-----
